.github/workflows/ci-cd.yml

```diff
@@ -129,3 +129,4 @@ jobs:
           max_attempts: 3
           retry_on: error
           command: docker push "theiaide/${{ matrix.IMAGE_NAME }}"
```

Comment:
I'm worried about the `docker push` command not being able to read the secrets, should we move to the official `docker-push` action?:

- https://docs.docker.com/ci-cd/github-actions/

```yaml
steps:
    
      - name: Check Out Repo 
        uses: actions/checkout@v2

      - name: Login to Docker Hub
        uses: docker/login-action@v1
        with:
          username: ${{ secrets.DOCKER_HUB_USERNAME }}
          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}

      - name: Set up Docker Buildx
        id: buildx
        uses: docker/setup-buildx-action@v1

      - name: Build and push
        id: docker_build
        uses: docker/build-push-action@v2
        with:
          context: ./
          file: ./Dockerfile
          push: true
          tags: ushamandya/simplewhale:latest

      - name: Image digest
        run: echo ${{ steps.docker_build.outputs.digest }}
```

Reply:
+ I believe `docker push` works if the login step passes, [same in this workflow](https://github.com/DukeNgn/minimal-docker-image/blob/main/.github/workflows/ci-cd.yml#L31-L38)
+ `docker/build-push-action@v2` does 2 things, **build** and **push** while we only need **push** in our use-case. It's a good idea tho, to replace our current build script (`build_container.sh`) with this action, but for the scope of this PR, I think we should only make changes to the login step.

Reply:
@DukeNgn sounds good - I'm ok with the minimal change, if it works